rllib/utils/metrics/stats.py

```diff
@@ -415,44 +436,72 @@ def merge_in_parallel(self, *others: "Stats") -> None:
                 into `self, meaning with equal weighting as the existing values in
                 `self`.
         """
-        # Make sure `others` have same reduction settings.
-        for o in others:
-            assert (
-                self._reduce_method == o._reduce_method
-            ), f"Mismatch beweteen stats {str(self)} (reduce_method={self._reduce_method}) and {str(o)} (reduce_method={o._reduce_method})"
-            assert (
-                self._ema_coeff == o._ema_coeff
-            ), f"Mismatch beweteen stats {str(self)} (ema_coeff={self._ema_coeff}) and {str(o)} (ema_coeff={o._ema_coeff})"
-
         win = self._window or float("inf")
 
+        # If any of the value lists have a length of 0 or if there is only one value and it is nan, we skip
+        stats_to_merge = [
+            s
+            for s in [self, *others]
```

Comment:
Why do we have to check for NaNs? Don't our default reduce methods (e.g. `np.nanmean()`) already take care of these?

Reply:
The problem is a tiny bit of error that I uncovered with the new tests.
Say you have two incoming stats to aggregate v1=[1] and v2=[np.nan]
Merging with summation over them with `reduce_per_index_on_parallel_merge=False` will now lead to
`result=[0.5, 0.5]`. But you are really merging only a single value, so it should be `result=[1]`. Other reduction methods `reduce_per_index_on_aggregate` applies have similar patterns.